src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs

```diff
@@ -22,7 +29,54 @@ public static void Initialize()
             ImageAnalytics.Initialize();
         }
 
+        private static unsafe ISchema GetModelSchema(IExceptionContext ectx, TFGraph graph)
+        {
+            long pos = 0;
+            TF_Operation oper;
+            var res = new List<KeyValuePair<string, ColumnType>>();
+            while ((oper = TFGraph.TF_GraphNextOperation(graph.handle, &pos)) != IntPtr.Zero)
+            {
+                var name = TFGraph.TF_OperationName(oper);
+                var type = TFGraph.TF_OperationOpType(oper);
+                var numOutputs = TFGraph.TF_OperationNumOutputs(oper);
+                if (numOutputs != 1)
+                    continue;
+
+                var numInputs = TFGraph.TF_OperationNumInputs(oper);
+                if (numInputs == 0)
```

Comment:
>numInputs [](start = 20, length = 9)

Why does `numInputs == 0` indicates? The input node does not have any input I believe???

Reply:
The input has OpType "Placeholder". There are other nodes with numInputs==0, which have OpType "Const". I am not sure what they do but I think we don't want them in our output schema. What do you think?

---
In reply to: [216508337](https://github.com/dotnet/machinelearning/pull/862#discussion_r216508337) [](ancestors = 216508337)

Reply:
What do you think if such nodes are not filter? Do you foresee adverse effect of doing this or not doing this?

---
In reply to: [216758114](https://github.com/dotnet/machinelearning/pull/862#discussion_r216758114) [](ancestors = 216758114,216508337)

Reply:
I added the nodes with 0 inputs to the schema as well. 

---
In reply to: [216817673](https://github.com/dotnet/machinelearning/pull/862#discussion_r216817673) [](ancestors = 216817673,216758114,216508337)